app/services/populate_topics.rb

```diff
@@ -15,8 +15,19 @@ def existing_topics
   private
 
   def add_to_kafka(topics_to_add)
-    topics_to_add.each do |topic|
-      OxKafka.instance.create_topic(topic)
+    topics_to_add.each do |topic_name|
+      # replication happens across brokers.  So, calculate the min of # brokers and replication factor
+      replication = [OxKafka.instance.brokers.length, topics_config.replication_by_topic[topic_name]].min
+
+      OxKafka.instance.create_topic(
+        topic_name,
+        replication_factor: replication,
+        num_partitions: replication
```

Comment:
I'm not sure why this is set to the replication factor.  I don't think they are related?

Reply:
Yeah dig into this some more, we'll need to be able to set those independently, soon.

Reply:
Also, min() vs # brokers is fine, but should log something when that happens - it should never happen in production, but is fine for dev and functionality testing.

Partitions should min vs. brokers, replication should min vs. partitions - since replication happens over partitions, you can't have more replicas than partitions.